### ISSUE TITLE
Corrupted images on Vukan backend

### ISSUE THREAD

**Issue by wbruna** [leejet/stable-diffusion.cpp#439]
I'm getting either corrupted or inconsistent images on Vulkan backend, for any resolution other than 512x512.

My system is a Linux PC with a Ryzen 3400G, almost-vanilla Debian 12 (using the distro graphic stack). All following tests with:
`--type f16 --lora-model-dir ./LoRA --model ./SD/dreamshaper_8.safetensors --prompt 'a fantasy character, detailed background, colorful<lora:lcm-lora-sdv1-5:1>' --cfg-scale 1.0 --sampling-method lcm --steps 4 --rng cuda --seed 42 -b 1 --color` 
, and a script alternating resolution and compiled binary (Vulkan or CPU backend).

320x512:

Vulkan 1 |  Vulkan 2 | CPU
:----:|:----:|:----:
![320x512_vulkan1](https://github.com/user-attachments/assets/a4d15af4-4440-460f-b285-da8b10b2e083) | ![320x512_vulkan2](https://github.com/user-attachments/assets/53573927-a2d0-4e4c-a43e-05a93fc6a593) | ![320x512_cpu](https://github.com/user-attachments/assets/da803094-e2f1-40c4-91ab-15bb5fd728f3)

Vulkan images look ok-ish (for such a small resolution anyway), but the same seed should produce the same image. And the CPU render looks quite different.

Second test, 384x384; similar behavior (changes between Vulkan 1 and 2 may not be apparent on the thumbnail):

Vulkan 1 |  Vulkan 2 | CPU
:----:|:----:|:----:
![384x384_vulkan1](https://github.com/user-attachments/assets/fd4487fa-50ac-43b3-851b-a468584a996c) | ![384x384_vulkan2](https://github.com/user-attachments/assets/e43984f8-4d57-4177-bdd9-02847babfdb0) | ![384x384_cpu](https://github.com/user-attachments/assets/b000997a-a473-483f-aaa2-c63a88ca8957)

The third test, 448x448, gets weird:

Vulkan 1 |  Vulkan 2 | CPU
:----:|:----:|:----:
![448x448_vulkan1](https://github.com/user-attachments/assets/3b2122dc-aeb2-43c6-bf7e-525839e3f1a4) | ![448x448_vulkan2](https://github.com/user-attachments/assets/c059fa75-99a6-422e-b363-a77b3a967e1f) | ![448x448_cpu](https://github.com/user-attachments/assets/3e1bb803-b592-42c3-bbe1-ce54d929bdfc)

At first, I blamed my PC drivers. But then, the 512x512 test:

Vulkan 1 |  Vulkan 2 | CPU
:----:|:----:|:----:
![512x512_vulkan1](https://github.com/user-attachments/assets/50333389-ea96-47e7-bffe-45bc64ca5d97) | ![512x512_vulkan2](https://github.com/user-attachments/assets/425cfbae-b5fc-4000-8cb3-131752422e31) | ![512x512_cpu](https://github.com/user-attachments/assets/30ceea89-5ba9-476b-a8a4-69abe5fa5a5d)

Looks absolutely fine, and identical between Vulkan and CPU.

In summary:
- 512x512 works fine;
- any other resolution produces inconsistent images between runs;
- some resolutions introduce artifacts

(related: #122 )

**Comment by stduhpf** [comment]
Good catch. Using the same prompt, i get a smilar behavior as you.
 Though I don't get anything as dramatic as your 448x448 results. (I only get variations of your "Vulkan 1" images, no matter how many times I try).
 
 I might try to investigate what's going on, but I'm not confident I'll figure it out.

**Comment by stduhpf** [comment]
@0cc4m Do you have a clue?

**Comment by stduhpf** [comment]
This also happens with images bigger than 512x512 if the resolution isn't a multiple of 128...

**Comment by 0cc4m** [comment]
Thank you for the detailed report, I'll look into it when I find some time.

**Comment by jeffbolznv** [comment]
I used GGML_VULKAN_CHECK_RESULTS to narrow down that group_norm was failing, reproduced it with a backend test, and pushed a fix to https://github.com/ggerganov/llama.cpp/pull/10496.

BTW, GGML_VULKAN_CHECK_RESULTS is really helpful, but it looks like it may get harder to build with the recent backend split.

**Comment by 0cc4m** [comment]
> BTW, GGML_VULKAN_CHECK_RESULTS is really helpful, but it looks like it may get harder to build with the recent backend split.

Yeah, I built it to narrow down these kinds of model issues. I hope we can keep it around.

**Comment by wbruna** [comment]
I manually applied 56d8a95 on a local build, and it seems to fix this issue. Thanks!

**Comment by wbruna** [comment]
This was fixed in ggml-org/ggml@38433280 ( 195d170 ).
